From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sevenrock <sevenrock@hotmail.de>
Date: Tue, 17 Dec 2024 21:05:13 +0100
Subject: [PATCH] msm: camera: tfe: Fix enum conversion warnings

../../../../../../kernel/motorola/sm6375/techpack/camera/drivers/cam_isp/isp_hw_mgr/cam_tfe_hw_mgr.c:3039:21:
 error: comparison of different enumeration types ('enum cam_tfe_csid_halt_cmd' and 'enum cam_ife_csid_halt_cmd')
 [-Werror,-Wenum-compare]
 3039 |         if (csid_halt_type == CAM_CSID_HALT_IMMEDIATELY)

Change-Id: Id4beb655f3aa2e1192dd434bcaec678d1e406777
Signed-off-by: Sevenrock <sevenrock@hotmail.de>
---
 techpack/camera/drivers/cam_isp/isp_hw_mgr/cam_tfe_hw_mgr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/techpack/camera/drivers/cam_isp/isp_hw_mgr/cam_tfe_hw_mgr.c b/techpack/camera/drivers/cam_isp/isp_hw_mgr/cam_tfe_hw_mgr.c
index e5342bd95163..7292a5899eca 100644
--- a/techpack/camera/drivers/cam_isp/isp_hw_mgr/cam_tfe_hw_mgr.c
+++ b/techpack/camera/drivers/cam_isp/isp_hw_mgr/cam_tfe_hw_mgr.c
@@ -3036,7 +3036,7 @@ static int cam_tfe_mgr_stop_hw(void *hw_mgr_priv, void *stop_hw_args)
 		master_base_idx = ctx->base[0].idx;
 
 	/*Change slave mode*/
-	if (csid_halt_type == CAM_CSID_HALT_IMMEDIATELY)
+	if ((enum cam_ife_csid_halt_cmd)csid_halt_type == CAM_CSID_HALT_IMMEDIATELY)
 		cam_tfe_mgr_csid_change_halt_mode(ctx,
 			CAM_TFE_CSID_HALT_MODE_INTERNAL);
 
-- 
2.39.5

